### ISSUE TITLE
Rethink how to handle a[href] > button cases

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#5184]
The following scenario fails the `target-size` rule. Axe-core allows nested controls inside links, because the HTML spec does as well. But in a scenario like this where the button is the only content of the link, and the target area is almost the same I think we can reasonably conclude this should fail the `nested-interactive` rule rather than the `target-size one:

## Expected

Axe-core report the following as a `nested-interactive` issue, rather than a `target-size` issue.

```html
<a href="#" style="padding: 0.2em 0.5em">
  <button>Click me</button>
</a>
```
